R/method-correlation.R

```diff
@@ -0,0 +1,100 @@
+#' Correlation
+#'
+#' This function returns correlation coefficients for all abundance variables vs all continuous sample metadata variables.
+#' Soon this function will be expanded to take multiple abundance datasets, and to run all-to-all correlations
+#' for abundance variables.
+#' 
+#' @param data AbundanceData object
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @import data.table
+#' @useDynLib microbiomeComputations
+#' @export
+setGeneric("correlation",
+  function(data, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) standardGeneric("correlation"),
+  signature = c("data")
+)
+
+#'@export
+setMethod("correlation", signature("AbundanceData"), function(data, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
+    df <- data@data
+    recordIdColumn <- data@recordIdColumn
+    naToZero <- data@imputeZero
+    ancestorIdColumns <- data@ancestorIdColumns
+    allIdColumns <- c(recordIdColumn, ancestorIdColumns)
+    sampleMetadata <- data@sampleMetadata
+
+    ## Initialize and check inputs
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    # Check that incoming df meets requirements
+    if (!'data.table' %in% class(df)) {
+      data.table::setDT(df)
+    }
+
+    # Check that incoming metadata meets requirements
+    if (!'data.table' %in% class(sampleMetadata)) {
+      data.table::setDT(sampleMetadata)
+    }
+
+    computeMessage <- ''
+    veupathUtils::logWithTime(paste("Received df table with", nrow(df), "samples and", (ncol(df)-1), "taxa."), verbose)
+
+    # Replace NA values with 0
+    if (naToZero) {
+      veupathUtils::setNaToZero(df)
+      veupathUtils::logWithTime("Replaced NAs with 0", verbose)
+    }
+
+    # Check that the order of sample ids match. If not, reorder so they do
+    if (!identical(df[[recordIdColumn]], sampleMetadata[[recordIdColumn]])) {
+      # Reorder sampleMetadata so that it matches the order in df
+      sampleMetadata <- sampleMetadata[match(df[[recordIdColumn]],sampleMetadata[[recordIdColumn]]), ]
+      veupathUtils::logWithTime("Reordered sampleMetadata rows based on abundance data sample id order.", verbose)
+    }
+
+    ## Find numeric metadata
+    numericMetadataCols <- veupathUtils::findNumericCols(sampleMetadata)
+    if (length(numericMetadataCols) < 1) {
+      stop("correlation requires at least one continuous metadata variable.")
```

Comment:
we should be careful here. were checking for numeric and then claiming we need continuous. dates are also continuous and some numbers are discrete.. id prefer, if we can, to use our variable/ collections metadata to identify whether we have appropriate data for this..

Reply:
good call. Updated in recent commits